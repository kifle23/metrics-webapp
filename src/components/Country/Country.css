import styled from 'styled-components';

export const CountryWrapper = styled.div`
`;
