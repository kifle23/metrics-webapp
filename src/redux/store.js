import { configureStore } from '@reduxjs/toolkit';
import weatherReducer from './weatherSlice';

export default configureStore({
  reducer: {
    data: weatherReducer,
  },
});