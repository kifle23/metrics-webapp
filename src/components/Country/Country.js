import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './Country.css';
import { getCities } from '../../redux/cities';

const Country = (props) => {
  const {
    name, number, vector, cities,
  } = props;
  const map = { background: `url(${vector}) no-repeat center center/cover` };
  const dispatch = useDispatch();

  return (
    <Link to="/cities" onClick={() => dispatch(getCities(cities))} className={styles.wrapper}>
      <i className="right_arrow" />
      <i style={map} className={styles.map} />
      <div className={styles.info}>
        <h3>{name}</h3>
        <span>{number}</span>
      </div>
    </Link>
  );
};

export default Country;

Country.propTypes = {
  name: PropTypes.string.isRequired,
  number: PropTypes.number.isRequired,
  vector: PropTypes.string.isRequired,
  cities: PropTypes.instanceOf(Array).isRequired,
};
