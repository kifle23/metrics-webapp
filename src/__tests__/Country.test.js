import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import Country from '../components/Country/Country';

describe('Country component', () => {
  test('renders correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <Country name="test" number={5} vector={null} cities={[]} />
        </Router>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
