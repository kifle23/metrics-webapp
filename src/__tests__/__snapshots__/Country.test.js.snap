// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country component renders correctly 1`] = `
<a
  className="wrapper"
  href="/cities"
  onClick={[Function]}
>
  <i
    className="right_arrow"
  />
  <div
    className="map"
    style={
      Object {
        "background": "url(null) no-repeat center center/cover",
      }
    }
  />
  <div
    className="info"
  >
    <h3>
      test
    </h3>
    <span>
      5
    </span>
  </div>
</a>
`;
